keg/utils.py

```diff
@@ -107,12 +107,13 @@ def pymodule_fpaths_to_objects(fpaths):
     """
     retval = []
     for fpath in fpaths:
-        path = pathlib.Path(fpath)
-        if path.exists():
+        try:
             pymodule_globals = {}
             with open(fpath) as fo:
                 exec(compile(fo.read(), fpath, 'exec'), pymodule_globals)
             retval.append((fpath, pymodule_globals))
+        except (FileNotFoundError, PermissionError):
```

Comment:
I see two exceptions but a test for only one of them.

Also, there are several more [exceptions listed related to file IO](https://docs.python.org/3/library/exceptions.html#os-exceptions) that could possible be triggered here?  Maybe just catch OSError which is the parent exception?

Finally, it might be worth looking at where this function is used.  I wonder if we should add `(fpath, None)` to `retval` to indicate that module had an error and couldn't be loaded.

Reply:
This function is used for config file lookups in keg. The FileNotFoundError is tested in the existing test for this function when it is passed a non-existent path. I had to use a mock to explicitly throw a PermissionError in the test I added.

I'm a bit reluctant to silently suppress any possible OSError; my goal was to skip errors we expect and don't care about. Specifically, some of the paths we are passed may not exist, or are in folders like `~/.config` which may not be readable to the keg process.

Reply:
- IsADirectoryError is the other OSError that could be relevant here, i.e. the path passed in happens to be a folder instead of a file. I'm fine with trapping those three individually and testing for each.
- agreed on needing a test for the file not existing. I mean, that test should have been there before since that case was handled... But we should have it now to verify.
- and yes, I think these results get used to report config paths via some of the CLI commands. So if a path fails for some reason, we should report that back somehow rather than silently ignoring the path. I.e. a developer may put a config in `~/.config` with proper permissions on the file, but if the folder is not searchable to the user, a difficult-to-troubleshoot condition will result unless we report the permissionerror encountered on that path. See the `develop config` command.